packages/backend-for-frontend/src/api/delegationApiConverter.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  bffApi,
+  catalogApi,
+  delegationApi,
+  tenantApi,
+} from "pagopa-interop-api-clients";
+import {
+  DelegationKind,
+  delegationKind,
+  delegationState,
+  DelegationState,
+} from "pagopa-interop-models";
+import { match } from "ts-pattern";
+
+export type DelegationsQueryParams = {
+  delegatorIds?: string[];
+  delegateIds?: string[];
+  delegationStates?: delegationApi.DelegationState[];
+  kind?: delegationApi.DelegationKind;
+  eserviceIds?: string[];
+};
+
+export function toDelegationState(
+  state: DelegationState
+): bffApi.DelegationState {
+  return match(state)
+    .with(delegationState.active, () => "ACTIVE" as const)
+    .with(delegationState.rejected, () => "REJECTED" as const)
+    .with(delegationState.revoked, () => "REVOKED" as const)
+    .with(
+      delegationState.waitingForApproval,
+      () => "WAITING_FOR_APPROVAL" as const
```

Comment:
```suggestion
    .with(delegationState.active, () => bffApi.DelegationState.Values.ACTIVE)
    .with(
      delegationState.rejected,
      () => bffApi.DelegationState.Values.REJECTED
    )
    .with(delegationState.revoked, () => bffApi.DelegationState.Values.REVOKED)
    .with(
      delegationState.waitingForApproval,
      () => bffApi.DelegationState.Values.WAITING_FOR_APPROVAL
```
IMHO
I agree it's more verbose but we could use the already defined strings, it's not a strong opinion